core/src/contracts/update.rs

```diff
@@ -41,26 +42,58 @@ pub enum UpdateContractError {
     ErgoTreeConstant(ErgoTreeConstantError),
     #[error("update contract: TryExtractFrom error {0:?}")]
     TryExtractFrom(TryExtractFromError),
+    #[error("contract error: {1:?}, expected P2S: {0}")]
+    WrappedWithExpectedP2SAddress(String, Box<Self>),
 }
 
-pub struct UpdateContractInputs<'a> {
-    pub contract_parameters: &'a UpdateContractParameters,
-    pub pool_nft_token_id: &'a TokenId,
-    pub ballot_token_id: &'a TokenId,
+#[derive(Debug, Clone)]
+pub struct UpdateContractInputs {
+    contract_parameters: UpdateContractParameters,
+    pub pool_nft_token_id: TokenId,
+    pub ballot_token_id: TokenId,
 }
 
-impl<'a> From<UpdateBoxWrapperInputs<'a>> for UpdateContractInputs<'a> {
-    fn from(wrapper_inputs: UpdateBoxWrapperInputs) -> UpdateContractInputs {
-        UpdateContractInputs {
-            contract_parameters: wrapper_inputs.contract_parameters,
-            pool_nft_token_id: wrapper_inputs.pool_nft_token_id,
-            ballot_token_id: wrapper_inputs.ballot_token_id,
-        }
+impl UpdateContractInputs {
+    pub fn create(
+        contract_parameters: UpdateContractParameters,
+        pool_nft_token_id: TokenId,
+        ballot_token_id: TokenId,
+    ) -> Result<Self, UpdateContractError> {
+        let network_prefix = contract_parameters.p2s.network();
+        let inputs_to_create_contract = Self {
+            contract_parameters,
+            pool_nft_token_id,
+            ballot_token_id,
+        };
+        let refresh_contract = UpdateContract::create(&inputs_to_create_contract)?;
+        let new_parameters = refresh_contract.parameters(network_prefix);
```

Comment:
```suggestion
        let update_contract = UpdateContract::create(&inputs_to_create_contract)?;
        let new_parameters = update_contract.parameters(network_prefix);
```

Reply:
Thanks for spotting this! Copy-pasting :facepalm: